### ISSUE TITLE
Canvases are still heckin' broke

### ISSUE THREAD

**Issue by icefoxen** [ggez/ggez#390]
Draws upside-down

Also doesn't draw the `Text` objects where you tell it to.

```rust
extern crate ggez;
use ggez::*;
use ggez::event;
use ggez::graphics::{Point2};

use std::env;
use std::path;

struct MainState {
    symbol_map: graphics::Image,
}


impl MainState {
    fn new(_ctx: &mut Context) -> GameResult<MainState> {
        let sym_map_canvas = graphics::Canvas::new(_ctx, 512, 512, conf::NumSamples::One)?;
        // let font = graphics::Font::default_font()?;
        let font = graphics::Font::new(_ctx, path::Path::new("/terminus.ttf"), 32)?;
        {

            graphics::set_canvas(_ctx, Some(&sym_map_canvas));

            graphics::clear(_ctx);

            let text = graphics::Text::new(_ctx, "A", &font)?;
            graphics::draw(_ctx, &text, Point2::new(0.0, 0.0), 0.0)?;

            let text = graphics::Text::new(_ctx, "X", &font)?;
            graphics::draw(_ctx, &text, Point2::new(64.0, 0.0), 0.0)?;
            graphics::set_canvas(_ctx, None);
        }
        let symbol_map = sym_map_canvas.into_inner();

        let s = MainState {
            symbol_map: symbol_map,
        };

        Ok(s)
    }
}

impl event::EventHandler for MainState {
    fn update(&mut self, _ctx: &mut Context) -> GameResult<()> {
        Ok(())
    }

    fn key_down_event(&mut self, ctx: &mut Context, keycode: event::Keycode, _keymod: event::Mod, _repeat: bool) {
        if keycode == event::Keycode::Q && _keymod & event::LSHIFTMOD == event::LSHIFTMOD {
            ctx.quit().expect("Should never fail");
        }
    }

    fn draw(&mut self, ctx: &mut Context) -> GameResult<()> {
        graphics::set_background_color(ctx, graphics::Color::from((0, 0, 0, 255)));
        graphics::clear(ctx);
        graphics::draw_ex(ctx, &self.symbol_map, graphics::DrawParam{
            src: graphics::Rect{x: 0.0, y: 0.0, h: 0.0625, w: 0.0625},
            dest: Point2::new(64.0, 64.0),
            .. Default::default()
        })?;

        graphics::draw_ex(ctx, &self.symbol_map, graphics::DrawParam{
            src: graphics::Rect{x: 0.0, y: 0.0, h: 1.0, w: 1.0},
            dest: Point2::new(128.0, 0.0),
            .. Default::default()
        })?;
        graphics::present(ctx);
        Ok(())
    }
}

pub fn main() {
    let c = conf::Conf::new();
    let ctx = &mut Context::load_from_conf("super_simple", "ggez", c).unwrap();

    if let Ok(manifest_dir) = env::var("CARGO_MANIFEST_DIR") {
        let mut path = path::PathBuf::from(manifest_dir);
        path.push("resources");
        ctx.filesystem.mount(&path, true);
    }

    let state = &mut MainState::new(ctx).unwrap();
    event::run(ctx, state).unwrap();
}

```

**Comment by stachetheoverlord** [comment]
Shouldn't this be closed?

**Comment by icefoxen** [comment]
I would hope so, but I don't remember what the fix was so I have to double-check.

**Comment by svenknobloch** [comment]
Was there ever a fix for this? I'm running into the flipped image issue (OSX Mojave).

**Comment by IGBC** [comment]
fixed in #936. More details about canvases are being dealt with in #813  and  #694. If this problem returns please make a new issue.
